src/vcard/mod_vcard.erl

```diff
@@ -458,42 +456,46 @@ unsafe_set_vcard(HostType, From, VCARD) ->
     end.
 
 
--spec set_vcard(HandlerAcc, HostType, From, VCARD) -> Result when
-      HandlerAcc :: ok | error(),
-      HostType :: mongooseim:host_type(),
-      From ::jid:jid(),
-      VCARD :: exml:element(),
-      Result :: ok | error().
-set_vcard(ok, _HostType, _From, _VCARD) ->
+-spec set_vcard(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: ok | {error, term()},
+    Params :: map(),
+    Extra :: map().
+set_vcard(ok, _Params, _Extra) ->
     ?LOG_DEBUG(#{what => hook_call_already_handled}),
-    ok;
-set_vcard({error, no_handler_defined}, HostType, From, VCARD) ->
-    try unsafe_set_vcard(HostType, From, VCARD) of
-        ok -> ok;
-        {error, Reason} ->
-            ?LOG_ERROR(#{what => unsafe_set_vcard_failed, reason => Reason}),
-            {error, Reason}
-    catch
-        E:R:S -> ?LOG_ERROR(#{what => unsafe_set_vcard_failed, class => E,
-                              reason => R, stacktrace => S}),
-               {error, {E, R}}
-    end;
-set_vcard({error, _} = E, _HostType, _From, _VCARD) -> E.
-
--spec remove_domain(mongoose_hooks:simple_acc(),
-                    mongooseim:host_type(), jid:lserver()) ->
-    mongoose_hooks:simple_acc().
-remove_domain(Acc, HostType, Domain) ->
+    {ok, ok};
+set_vcard({error, no_handler_defined}, #{user := From, vcard := VCARD}, #{host_type := HostType}) ->
+    Result = try unsafe_set_vcard(HostType, From, VCARD) of
+                 ok -> ok;
+                 {error, Reason} ->
+                     ?LOG_ERROR(#{what => unsafe_set_vcard_failed, reason => Reason}),
+                     {error, Reason}
+             catch
+                 E:R:S -> ?LOG_ERROR(#{what => unsafe_set_vcard_failed, class => E,
+                                     reason => R, stacktrace => S}),
+                     {error, {E, R}}
+             end,
+    {ok, Result};
+set_vcard({error, _} = Error, _Params, _Extra) ->
+    {ok, Error}.
+
+-spec remove_domain(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: mongoose_domain_api:remove_domain_acc(),
+    Params :: map(),
+    Extra :: map().
+remove_domain(Acc, #{domain := Domain}, #{host_type := HostType}) ->
     mod_vcard_backend:remove_domain(HostType, Domain),
-    Acc.
+    {ok, Acc}.
 
-%% #rh
-remove_user(Acc, User, Server) ->
+-spec remove_user(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: mongoose_acc:t(),
+    Params :: map(),
+    Extra :: map().
+remove_user(Acc, #{jid := #jid{luser = User, lserver = Server}}, #{host_type := HostType}) ->
     HostType = mongoose_acc:host_type(Acc),
```

Comment:
I think we can remove this line since HostType is already bound in function's head.